server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -157,10 +157,15 @@
         for (ExpressionResolver expressionResolver : expressionResolvers) {
             expressions = expressionResolver.resolve(context, expressions);
         }
-
+        
         if (expressions.isEmpty()) {
             if (!options.allowNoIndices()) {
                 IndexNotFoundException infe = new IndexNotFoundException((String)null);
+                if (indexExpressions.length == 1) {
+                    if (indexExpressions[0].equals(MetaData.ALL)) {
+                        infe = new IndexNotFoundException("no indices exist", (String)null);
+                    }
+                }
```

Comment:
nit: what about initializing the exception once in both cases like we do below?

Reply:
I didn't see what was done below, but is this what you mean <br>
<code>IndexNotFoundException noIndicesException = new IndexNotFoundException("no indices exist", (String)null); </code> <br>
and throwing it in both cases.

Reply:
I mean to have an if with two branches, and instantiate a different exception on the two branches, rather than re-initializing in one case. You have done this yourself in the other change you have below.

Reply:
alright

Reply:
<code>IndexNotFoundException infe; </code>
 <code>              if (indexExpressions.length == 1) { </code>
<code>                    if (indexExpressions[0].equals(MetaData.ALL)) { </code>
 <code>                       infe = new IndexNotFoundException("no indices exist", (String)null); </code>
 <code>                   }</code>
 <code>               } else { </code>
  <code>                  infe = new IndexNotFoundException((String)null);</code>
    <code>            }</code>
Something like this

Reply:
yes